modules/text/cmake/FindTesseract.cmake

```diff
@@ -3,24 +3,24 @@ if(COMMAND pkg_check_modules)
   pkg_check_modules(Tesseract tesseract lept)
 endif()
 if(NOT Tesseract_FOUND)
-  find_path(Tesseract_INCLUDE_DIR tesseract/baseapi.h
+  find_path(Tesseract_INCLUDE_DIRS tesseract/baseapi.h
```

Comment:
This change is wrong.

Reply:
Could you elaborate?

The idea is to avoid having both `Tesseract_INCLUDE_DIR` and `Tesseract_INCLUDE_DIRS` in the global namespace -- because confusing them is what I presume caused this bug in the first place. (Same for `_LIBRARY` vs `_LIBRARIES`.)

Reply:
Could you elaborate?

The idea is to avoid having both `Tesseract_INCLUDE_DIR` and `Tesseract_INCLUDE_DIRS` in the global namespace -- because confusing them is what I presume caused this bug in the first place. (Same for `_LIBRARY` vs `_LIBRARIES`.)

So to "have no effect, except names renames" is the intent.

Reply:
find_path() always find one entry. DIRS is plural noun.

```
find_path(Tesseract_INCLUDE_DIR ...)  # this goes into CMake cache and can be changed by user

set(Tesseract_INCLUDE_DIRS "${Tesseract_INCLUDE_DIR}" "${OtherDependency_INCLUDE_DIR}")  # DIRs variant (should not be cached)


# Usage
ocv_include_directories(${Tesseract_INCLUDE_DIRS})
```

Reply:
The same story is about `find_library()` / `LIBRARY` / `LIBRARIES`.

Reply:
So what if it finds one entry? We start with one, then add more further in the code if needed.

A more convincing argument is that there is a bug in my logic: if we take the value of a cache entry and add more stuff to it, that stuff will be added again on subsequent runs!

I considered `unset()`'ing `Tesseract_INCLUDE_DIR` et al. instead but this would cause it to be re-detected on each run.

I would stiil like to make the names more distinct this way or another and suggest that `Tesseract_INCLUDE_DIR` is not supposed to be used. I can't find any naming conventions for CMake for "private" cache entries (names starting with underscore that are typically used for this seem to rather be reserved for CMake's internal use here).

Reply:
Since the primary bug is already fixed, this is going to be done in a separate PR though.

Reply:
Variables in find_path()/find_library()/etc:
- may contain one path or "NOT_FOUND".
- values are cached between CMake runs (do not searching again and again).

> unset()'ing Tesseract_INCLUDE_DIR

Bad idea.
This blocks used to specify own path via command-line if auto-detection doesn't work properly (some new paths to check or custom location of dependency).